Signing methods

aaronlehmann opened the pull request:
Use RSA-PSS for all RSA signatures

Change the names of the signing methods to make consistent.

Use uppercase when referring to crypto and signing algorithms.

This incorporates the changes in the PR https://github.com/endophage/gotuf/pull/4

Commit:
Fix GenerateFileMeta tests

GenerateFileMeta was renamed to NewFileMeta and moved to the "data"
package. Update and move the corresponding tests.

Also, fix TestGenerateFileMetaDefault to specify sha512 as the hash
algorithm.

Commit:
Fix tests in "store" directory

* Pass a missing arg to NewHTTPStore

* Remove unused file store_test.go

* Pass the key to data.NewPublicKey in PEM format

* Fix a missing check that caused a crash when the key was not in the
  right format

* Create /tmp/sqlite before attempting to use a sqlite DB in that
  directory. Handle errors with a panic (before, errors would crash).

Commit:
Fix format string issues in Client's Update method

Commit:
Remove client_test.go

This test is based on updating from http://mirror1.poly.edu/test-pypi/.
Aside from the inconveniences of depending on external resources, the
format of these pypi files isn't compatible with the current gotuf, so
the test isn't workable.

Commit:
Change Mkdir permissions for sqlite db directory to 0755

Commit:
Merge branch 'master' into rsa-methods

Commit:
Use RSA-PSS for all RSA signatures

Change the names of the signing methods to make consistent.

Use uppercase when referring to crypto and signing algorithms.

endophage commented on signed/verifiers.go:
```diff
@@ -17,10 +17,9 @@ import (
-	"ed25519": Ed25519Verifier{},
-	"rsa":     RSAVerifier{},
-	"rsassa-pkcs1-v1_5-sign": RSAPemVerifier{}, // RSASSA-PKCS1-V1_5-SIGN
-	"pycrypto-pkcs#1 pss":    RSAPSSVerifier{},
```
Leave in this line and the line above for legacy and cross language compatibility reasons.

endophage replied:
Actually, maybe we don't need the rsassa... one. But definitely need to keep pycrypto to maintain compatibility with Python implementation.

endophage commented on signed/verifiers.go:
```diff
@@ -80,55 +81,34 @@ func (v RSAVerifier) Verify(key data.Key, sig []byte, msg []byte) error {
 	return nil
 }
 
-type RSAPemVerifier struct{}
```
~~This verifier and its Verify functions need to remain in their original format. They provide compatibility for the Python implementation of TUF.~~

(meant to put this on the RSAPSSVerifier)

endophage commented on signed/verifiers.go:
```diff
@@ -80,55 +81,34 @@ func (v RSAVerifier) Verify(key data.Key, sig []byte, msg []byte) error {
-}
-
-// RSAPSSVerifier checks RSASSA-PSS signatures
-type RSAPSSVerifier struct{}
```
This verifier and its Verify functions need to remain in their original format. They provide compatibility for the Python implementation of TUF.

Commit:
Reintroduce "pycrypto-pkcs#1 pss" signature algorithm

The only difference between this one and the new rsassa-pss-x509
algorithm is that "rsassa-pss-x509" doesn't get PEM-encoded input.

Factor out PSS verification code into a separate function.

Signed-off-by: Aaron Lehmann <aaron.lehmann@docker.com>
